tapdb/assets_store.go

```diff
@@ -2513,6 +2514,9 @@ func (a *AssetStore) ConfirmParcelDelivery(ctx context.Context,
 					"output: %w", err)
 			}
 
+			// TODO(roasbeef): asset version needed above?
```

Comment:
Is our intention to make V1 the new default? I though this would just be op-in for cases where we actually need the flexibility for certain constructions?

Reply:
If `V1` is the new default, can't we just make these changes as breaking changes in `V0`? Though I appreciate the new version handling logic. I guess we want to support existing assets.

Reply:
Yeah agree it's just about having this be opt-in, so fine that change isn't encumbered. 

> If V1 is the new default, can't we just make these changes as breaking changes in V0? 

The current implementation is a sort of a hedge, if it turns out that we were wrong, then we can disallow V1 in the verifier.